system/lib/wasm_worker/library_wasm_worker.c

```diff
@@ -17,6 +17,8 @@
 // Options:
 // #define STACK_OVERFLOW_CHECK 0/1/2 : set to the current stack overflow check mode
 
+// When pthread enabled initializing main thread tls has been done in emscripten_init_main_thread
+#ifndef __EMSCRIPTEN_PTHREADS__
```

Comment:
As far as I can tell we don't build with `-pthread` variant of this library.     We only have `_stub` and `-debug` flavors.   Would would need to add a `-mt` flavor too if you want this to work.

The other alternative would be to give this function and the pthread `_emscripten_tls_init` function the same name.. and then make this one weak?    What do you think @juj ?